packages/splitio-adapter/modules/adapter/adapter.js

```diff
@@ -54,13 +54,13 @@ const ensureUser = user => ({
   ...user,
 });
 
-const initializeClient = (authorizationKey, user, options) => {
+const initializeClient = (authorizationKey, key, options) => {
   const factory = splitio({
+    ...options,
```

Comment:
We avoid being open to overwrites by flipping the spreading.

Reply:
There is still one problem. If we want to add the option `core.labelsEnabled`, which is part of the `core` key, we cannot without redefining the full `core` object. Because the `autorizationKey` lives within.

How can we allow such customization without hassles for the user?

Reply:
Sorry missed this comment. I will have to think about it.

Reply:
Hm. So generally I would like to keep `adapterArgs` across adapters kind of alike. Which is a bit of a unachievable goal in of itself. However worth it maybe. I see kind of two options:

1. Make `adapterArgs` and hence the `initializeClient` receive only two args `core` and `options`
3. Make `options` allow to contain a `core` property itself which we spread into the `core` below
3. Keeping things as is while accepting a third argument in `core`
   - feels a bit wonky as you rightfully pointed out that `authoizationKey` and `key` are part of `core`

I slightly prefer options 1. and 2. with a slight preference to 2. A consumer of the adapter will never know that the first two properties are actually part of `core` and doesn't have too. While we keep the configuration objects shallow (in comparison how it would be when going with option 1.)

What's your take? And sorry again for the delay.

Reply:
Happy New Year! 🎉 

> A consumer of the adapter will never know that the first two properties are actually part of core and doesn't have too.

Slightly not true. If we want to configure our adapter we'll look at the docs and we know that the `key` field is part of `core`.

Anyway, let's go for option 2, this seems the best.

Reply:
🎈 Happy new year to you too 🚀 

Yeah I agree that me statement is slightly off. I just rather have adapter args look a like and it's easy to convey to users that options will just be passed to the adapter "correctly". Let me spin up my proposed solution as a PR then.

Reply:
Voila. This should do. Tested it a bit. 4ea62a5b6c033c33f408d4efb3d81913a5311b02